.travis.yml

```diff
@@ -97,26 +100,41 @@ jobs:
     - { rvm: *jruby_head, jdk: *jdk_8, env: *rails_master_jruby }
 
   exclude:
+    - { rvm: *ruby_2_4,  env: *rails_4_1 }
+    - { rvm: *ruby_2_5,  env: *rails_4_1 }
+    - { rvm: *ruby_2_6,  env: *rails_4_1 }
+    - { rvm: *ruby_2_7,  env: *rails_4_1 }
     - { rvm: *ruby_head, env: *rails_4_1 }
 
+    - { rvm: *ruby_2_7,  env: *rails_4_2 }
+    - { rvm: *ruby_head, env: *rails_4_2 }
+
     - { rvm: *ruby_2_1,  env: *rails_5_0 }
     - { rvm: *ruby_2_1,  env: *rails_5_1 }
     - { rvm: *ruby_2_1,  env: *rails_5_2 }
 
     - { rvm: *ruby_2_1,  env: *rails_6_0 }
     - { rvm: *ruby_2_2,  env: *rails_6_0 }
     - { rvm: *ruby_2_3,  env: *rails_6_0 }
+    - { rvm: *ruby_2_4,  env: *rails_6_0 }
 
     - { rvm: *ruby_2_1,  env: *rails_master }
     - { rvm: *ruby_2_2,  env: *rails_master }
     - { rvm: *ruby_2_3,  env: *rails_master }
+    - { rvm: *ruby_2_4,  env: *rails_master }
 
   allow_failures:
```

Comment:
I almost never look at the allowed failures. unless you are, there's really no reason to put the load on the CI

Reply:
I agree with you very much. The fact is, the build matrix, how its configured, runs tests for each Ruby version listed under `rvm` against each of the Rails versions under `jobs`
```yml
rvm:
  - *ruby_2_1
  - *ruby_2_2
  - *ruby_2_3
  - *ruby_2_4
  - *ruby_2_5
  - *ruby_2_6
  - *ruby_2_7
  - *ruby_head

jobs:
  - *rails_4_1
  - *rails_4_2
  - *rails_5_0
  - *rails_5_1
  - *rails_5_2
  - *rails_6_0
  - *rails_master
```

In order **to SKIP the CI**, we list the envs (rvm/jobs) under `exclude` section:
```diff
exclude:
  - { rvm: *ruby_2_4,  env: *rails_4_1 }
  - { rvm: *ruby_2_5,  env: *rails_4_1 }
  - { rvm: *ruby_2_6,  env: *rails_4_1 }
  - { rvm: *ruby_2_7,  env: *rails_4_1 }
  - { rvm: *ruby_head, env: *rails_4_1 }

  - { rvm: *ruby_2_7,  env: *rails_4_2 }
  - { rvm: *ruby_head, env: *rails_4_2 }

  - { rvm: *ruby_2_1,  env: *rails_5_0 }
  - { rvm: *ruby_2_1,  env: *rails_5_1 }
  - { rvm: *ruby_2_1,  env: *rails_5_2 }

  - { rvm: *ruby_2_1,  env: *rails_6_0 }
  - { rvm: *ruby_2_2,  env: *rails_6_0 }
  - { rvm: *ruby_2_3,  env: *rails_6_0 }
  - { rvm: *ruby_2_4,  env: *rails_6_0 }

  - { rvm: *ruby_2_1,  env: *rails_master }
  - { rvm: *ruby_2_2,  env: *rails_master }
  - { rvm: *ruby_2_3,  env: *rails_master }
+ - { rvm: *ruby_2_4,  env: *rails_master }
```

Moreover, the envs which are not run under default configuration (e.g. **jRuby**) have been listed under `include` section.

And the envs which are from (ruby/jruby)-head or rails-master, have been listed under `allow_failures`, so that we can see any upcoming errors in advance due to new changes on those envs

Reply:
here is the last successful build on Travis, where you can find all the envs that are using the CI.

https://travis-ci.org/rails-api/active_model_serializers/builds/604587180

let me know if we need to exclude any of the envs from the list

Reply:
yeah, that's why I removed some from the rvm list and selectively added them to the include list